datastore/importers/nmdb-import-systeminfo/nmdb-import-systeminfo.cpp

```diff
@@ -0,0 +1,121 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantiasl portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+// =============================================================================
+// Maintained by Sandia National Laboratories <Netmeld@sandia.gov>
+// =============================================================================
+
+#include <netmeld/datastore/tools/AbstractImportSpiritTool.hpp>
+#include "Parser.hpp"
+
+namespace nmdt = netmeld::datastore::tools;
+
+// =============================================================================
+// Import tool definition
+// =============================================================================
+template<typename P, typename R>
+class Tool : public nmdt::AbstractImportSpiritTool<P,R>
+{
+  // ===========================================================================
+  // Variables
+  // ===========================================================================
+  private:
+  protected:
+  public:
+
+  // ===========================================================================
+  // Constructors
+  // ===========================================================================
+  private:
+  protected:
+  public:
+    Tool() : nmdt::AbstractImportSpiritTool<P,R>
+      (
+       "systeminfo",           // command line tool imports data from
+       PROGRAM_NAME,           // program name (set in CMakeLists.txt)
+       PROGRAM_VERSION         // program version (set in CMakeLists.txt)
+      )
+    {}
+
+  // ===========================================================================
+  // Methods
+  // ===========================================================================
+  private: // Methods part of internal API
+    void
+    specificInserts(pqxx::transaction_base& t) override
+    {
+      const auto& toolRunId {this->getToolRunId()};
+      const auto& deviceId  {this->getDeviceId()};
+      for (auto& results : this->tResults) {
+
+        LOG_DEBUG << "Iterating over DeviceInformation\n";
+        results.devInfo.save(t, toolRunId, deviceId);
+        LOG_DEBUG << results.devInfo.toDebugString() << "\n";
+
+        LOG_DEBUG << "Iterating over OperatingSystems\n";
+        results.os.save(t, toolRunId, deviceId);
+        LOG_DEBUG << results.os.toDebugString() << "\n";
+
+        LOG_DEBUG << "Iterating over Hotfixes\n";
+        t.exec_prepared("insert_raw_hotfixes", toolRunId, ConvertVectorToPostgresArray(results.hotfixes));
```

Comment:
See my below comment, but this is much more complicated than needed.
```suggestion
        t.exec_prepared("insert_raw_hotfixes", toolRunId, results.hotfixes);
```

Reply:
Committed suggestion